robot.py

```diff
@@ -19,9 +21,12 @@ def __init__(self, config, logger):
             if driver_name == 'gps':
                 driver = GPS(config[driver_name], logger, output=self.input_gate,
                              name=driver_name)
-                self.drivers.append(driver)
+            elif driver_name == 'imu':
+                driver = IMU(config[driver_name], logger, output=self.input_gate,
+                             name=driver_name)
```

Comment:
Could package `drivers` export a dict of known drivers? In that case we could just index using `driver_name` and use that. No need to handle each one on a case by case basis.

Reply:
Good point - added refactoring as we discussed yesterday (tested on old GPS only).